src/GhSync/Program.cs

```diff
@@ -1,54 +1,62 @@
 ﻿using System.CommandLine;
 using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
-using System.CommandLine.Invocation;
 using System.CommandLine.Builder;
 using System.CommandLine.Parsing;
 
 namespace gh_sync;
 
 class Program
 {
+    protected readonly IServiceProvider services;
 
-    internal record PullIssueOptions(bool DryRun, bool AllowExisting);
-    private static Command PullIssueCommand()
+    public Program(Action<IServiceCollection>? configureServices = null)
     {
-
-        var command = new Command("pull-gh")
+        var services = new ServiceCollection();
+        if (configureServices == null)
         {
-            new Argument<string>(
-                "repo",
-                description: "GitHub repository to pull the issue from."
-            ),
+            new Startup().ConfigureServices(services);
+        }
+        else
+        {
+            configureServices(services);
+        }
+        this.services = services.BuildServiceProvider();
+    }
 
-            new Argument<int>(
-                "issue",
-                description: "ID of the issue to pull into ADO."
-            ),
+    static async Task<int> Main(string[] args) =>
+        await new Program().Invoke(args);
 
-            new Option(
-                "--dry-run",
-                description: "Don't actually pull the GitHub issue into ADO."
-            ),
 
-            new Option(
-                "--allow-existing",
-                description: "Allow pulling an issue into ADO, even when a work item or bug already exists for that issue."
-            )
+    internal record PullIssueOptions(bool DryRun, bool AllowExisting);
+    private Command PullIssueCommand()
+    {
+        var repo = new Argument<string>("repo", "GitHub repository to pull the issue from.");
+        var issue = new Argument<int>("issue", "ID of the issue to pull into ADO.");
+        var dryRun = new Option<bool>("--dry-run", "Don't actually pull the GitHub issue into ADO.");
+        var allowExisting = new Option<bool>("--allow-existing", "Allow pulling an issue into ADO, even when a work item or bug already exists for that issue.");
+
+        var command = new Command("pull-gh", "Pull from GitHub into ADO")
+        {
+            repo,
```

Comment:
Looks like the old behavior can be implemented by adding System.CommandLine.NamingConventionBinder: https://github.com/dotnet/command-line-api/issues/1537